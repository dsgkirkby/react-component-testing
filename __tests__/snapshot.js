const renderer = require('react-test-renderer');
// const TestUtils = require('react-addons-test-utils');
const Login = require('../src/login');
const React = require('react');

describe('Snapshot Testing', () => {
  it('renders correctly when there are no errors', () => {
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('updates the input values when changed', () => {
    const tree = renderer.create(<Login/>);

    // Change the input values - waiting on selector API for test renderer as mentioned in https://github.com/facebook/jest/issues/1411

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when there are errors', () => {
    const tree = renderer.create(<Login error={'test error'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
