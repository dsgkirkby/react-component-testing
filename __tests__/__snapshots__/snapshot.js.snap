exports[`Snapshot Testing renders correctly when there are errors 1`] = `
<div>
  <div>
    <label>
      Username:
    </label>
    <input
      onChange={[Function onChange]}
      onKeyDown={[Function onKeyDown]}
      value="" />
  </div>
  <div>
    <label>
      Password:
    </label>
    <input
      onChange={[Function onChange]}
      onKeyDown={[Function onKeyDown]}
      type="password"
      value="" />
  </div>
  <p
    className="error">
    test error
  </p>
</div>
`;

exports[`Snapshot Testing renders correctly when there are no errors 1`] = `
<div>
  <div>
    <label>
      Username:
    </label>
    <input
      onChange={[Function onChange]}
      onKeyDown={[Function onKeyDown]}
      value="" />
  </div>
  <div>
    <label>
      Password:
    </label>
    <input
      onChange={[Function onChange]}
      onKeyDown={[Function onKeyDown]}
      type="password"
      value="" />
  </div>
</div>
`;
